java-frontend/src/main/java/org/sonar/java/se/constraint/ConstraintsByDomain.java

```diff
@@ -92,6 +93,11 @@ public boolean equals(Object o) {
   }
 
   @Override
+  public String toString() {
+    return "[" + stream().map(Constraint::toString).sorted().collect(Collectors.joining(",")) + "]";
```

Comment:
Minor: you can provide suffix and prefix args to `Collectors.joining` and avoid adding brackets with `+` 

I really like this idea :)

Reply:
Didn't know that one. Nice!